src/VisualStudio/Core/Def/Implementation/FindUsages/VisualStudioFindSymbolMonikerUsagesService.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.Editor.SymbolMonikers;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using VS.IntelliNav.Contracts;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.FindUsages
+{
+    [ExportWorkspaceService(typeof(IFindSymbolMonikerUsagesService), layer: ServiceLayer.Host), Shared]
+    internal class VisualStudioFindSymbolMonikerUsagesService : AbstractFindSymbolMonikerUsagesService
+    {
+        private readonly ICodeIndexProvider? _codeIndexProvider;
+
+        [ImportingConstructor]
+        public VisualStudioFindSymbolMonikerUsagesService(
+            [Import(AllowDefault = true)] ICodeIndexProvider? codeIndexProvider)
```

Comment:
What happens here if there are two exports of this type? It shouldn't happen but if it would cause an exception it would be good to handle it.

Reply:
@BertanAygun MEF simply wouldn't be able to satisfy the part, and would reject us from the composition.

Reply:
Actually a correction: since Cyrus is doing the `AllowDefault = true` now it'll not reject but we'll just not do anything.